docs/source/release_notes/release-notes.rst

```diff
@@ -64,6 +64,11 @@ develop
            If this breaks your tests, you can fix it with making real schema for tests or by switching to AtlasDbConstants.GENERIC_TABLE_METADATA
            (`Pull Request <https://github.com/palantir/1925>`__)
 
+    *    - |userbreak| |fixed|
+         - Fixed a bug that caused Cassandra to always use the minimum compression block size of 4KB instead of the requested compression block size. Users must explicitly rewrite table metadata for any tables that requested explicit compression, as any tables that were created previously will not respect the compression block size in the schema. This can have a very large performance impact (both positive and negative in different cases), so users may need to remove the explicit compression request from their schema if this causes a performance regression.
```

Comment:
It isn't strictly required (things will still work), but not doing this will mean that old customers would expect different performance compared to new customers using the same schema.  Everyone that uses Cassandra and has explicitCompressionRequested or explicitCompressionBlockSizeKB in their schema is affected (or if they use the StreamStore since the stream TableDefinition sets this).  Considering that guidance on schemas has been to request explicit compression for tables that are primarily for range scans or tables where individual rows are actually large, people who use this where appropriate in their schema should expect an improvement overall (range scans should be faster while individual puts/gets should be slower if those requests are much smaller than the block size).  I don't have a good test setup to be able to quantify exactly how big of a difference it makes, as it will depend on hardware, the schema, and the pattern of reads/writes.  If customers wrote a schema with an explicit compression block size of 64k but it turns out that they primarily do much smaller reads, then they would expect to see worse performance after this change, in which case they should simply remove the explicit compression setting from their schema (delete the line) and rewrite table metadata.

Fixing this also can potentially reduce space usage in some cases (larger block sizes have the potential to compress better), but I think read/write performance is probably something people care more about than saving space.

The instructions for customers is exactly the same as if they were making any change to their schema -- a one-time upgrade task is required to explicitly rewrite the table metadata.  The internal product I can speak for has a standard way of doing this; I cannot speak for the other products that use AtlasDB.